### PR TITLE
network entity refactoring

### PR DESCRIPTION
**Pull request by John-Lin**
All Test cases have to modify against current changes

### CODE REVIEW

File: src/networkcontroller/network_controller_test.go [hunk 1]
```diff
@@ -100,88 +100,93 @@ func (suite *NetworkControllerTestSuite) TestNew() {
...
-
 	tName := namesgenerator.GetRandomName(0)
 	network := entity.Network{
-		Name: tName,
-		OVS: entity.OVSNetwork{
-			BridgeName:    tName,
-			PhysicalPorts: []entity.PhysicalPort{eth1},
+		Type:       entity.OVSKernelspaceNetworkType,
+		Name:       tName,
+		VLANTags:   []int32{0, 2048, 4095},
```
Comment (zyfdegh) [inline src/networkcontroller/network_controller_test.go]: Need double check: you're using `4095`, not `4096`?

File: src/networkprovider/fake.go [hunk 2]
```diff
@@ -7,27 +7,37 @@ import (
 	"github.com/linkernetworks/vortex/src/serviceprovider"
 )
 
-type FakeNetworkProvider struct {
-	entity.FakeNetwork
+type fakenetworkprovider struct {
```
Comment (zyfdegh) [inline src/networkprovider/fake.go]: fake**N**etwork**P**rovider

File: src/networkprovider/ovs_netdev.go [hunk 3]
```diff
@@ -10,107 +10,98 @@ import (
 	"gopkg.in/mgo.v2/bson"
 )
 
-type OVSUserspaceNetworkProvider struct {
-	entity.OVSUserspaceNetwork
+type userspacenetworkprovider struct {
```
Comment (zyfdegh) [inline src/networkprovider/ovs_netdev.go]: userspace**N**etwork**P**rovider

File: src/server/handler_network_test.go [hunk 4]
```diff
@@ -252,64 +276,67 @@ func (suite *NetworkTestSuite) TestGetNetworkWithInvalidID() {
...
-			for i, v := range retNetworks {
-				suite.Equal(networks[i].Name, v.Name)
-				suite.Equal(networks[i].Type, v.Type)
-				suite.Equal(networks[i].NodeName, v.NodeName)
-			}
-
-		})
-	}
-}
+// func (suite *NetworkTestSuite) TestListNetwork() {
```
Comment (zyfdegh) [inline src/server/handler_network_test.go]: Remove the comments, or use t.Skip()

File: src/networkcontroller/network_controller_test.go (hunk 1 above)
Comment (John-Lin) [reply src/networkcontroller/network_controller_test.go]: yes! 4096 is a invalid number

**Pull request merged**